pennylane/ops/op_math/decompositions/two_qubit_unitary.py

```diff
@@ -347,6 +347,7 @@ def _decomposition_2_cnots(U, wires):
     part has the same spectrum as U, and then we can recover A, B, C, D.
     """
     # Compute the rotation angles
+
     u = math.dot(Edag, math.dot(U, E))
```

Comment:
So these are the eigenvalues of a unitary?

Reply:
`evs` are eigenvalues. I'm not sure what `u` is.

Reply:
Without understanding more, I'm a little worried that this fix just moves the discontinuity to another point.

Reply:
I had a brief look into this (need something numerical on my plate), and the issue here appears to be values that differ greater than the macheps^2, which fails to register differences when doing calls in `_extract_su2su2_prefactors`. This means that data being fed into LAPACK calls (e.g. for the determinant calculation, which uses LU decomp) causes the permutation matrix to flip, and introduces a sign change that shouldn't be present when values are comparable at least within the macheps range. The answer here is to add 10*macheps and prevent this from happening, which is a perfectly normal thing to do when dealing with back to back LAPACK (aka numpy/scipy) calls, and the way most scientific computing applications handle this. If we were shipping this functionality as primitive, we wouln't need this, but since we want application-layer dependencies, this is the way. Anyway, my 2c

Reply:
So I've found a more specific place to place the 5 * precision, but still have no clue as to why that particular thing is causing an issue.